src/container/srv_target.c

```diff
@@ -1097,6 +1097,20 @@ cont_child_destroy_one(void *vin)
 static void
 cont_delete_ec_agg(uuid_t pool_uuid, uuid_t cont_uuid);
 
+/* Destroys the container, but only on this xstream */
+int
+ds_cont_tgt_destroy_this_xstream(uuid_t pool_uuid, uuid_t cont_uuid)
+{
+	struct cont_tgt_destroy_in in;
+
+	uuid_copy(in.tdi_pool_uuid, pool_uuid);
+	uuid_copy(in.tdi_uuid, cont_uuid);
+
+	cont_delete_ec_agg(pool_uuid, cont_uuid);
```

Comment:
This line is my only concern here - disabling aggregation for this container. It looks like this operates node-wide, but I think this is needed.

Reply:
I didn't look into the  detail yet, but I'm wondering when do you re-enable it?

Reply:
No idea. The original ds_cont_tgt_destroy doesn't re-enable it either. I assume it gets re-enabled when the container is re-created?

Reply:
This is just a copy/paste of the function just below it (expand in github) except doesn't use dss_thread_collective, so it just runs on this one thread.

Reply:
The original ds_cont_tgt_destroy() is to do a real container destroy, but here you just want to delete existing data, the container isn't destroyed, so I suppose you need to re-enable it somehow?

Reply:
@wangdi1 I need your help to figure out what to do here. 

Currently (master), reintegration will delete containers that are being reintegrated on **all** xstreams for a rank - not just the ones that are being reintegrated. This causes data loss when only one target is excluded / reintegrated. 

This patch changes that behavior to only delete the container on the specific target that is being reintegrated. We definitely need to do this. 

The problem is that I don't know what needs to be done for aggregation. In the container destroy function `ds_cont_tgt_destroy` it calls `cont_delete_ec_agg` prior to destroying the container, but I don't understand what this is doing. It just removes the container from a list. This seems to affect all xstreams. 

If I want to destroy the container on just one target, what should I do? Is it reasonable to call this same API?

Reply:
It looks to me this cont_delete_ec_agg() just removes the container from a list so that the EC aggregation epoch boundary query ULT won't query the container anymore.

So I don't think cont_delete_ec_agg() should be called here.

Reply:
Sounds fine to me... would still like @wangdi1's opinion though just in case this (or something like it) is needed to delete a container from a single target.

Reply:
yes, you do not need call cont_delete_ec_agg() here, which is only needed if one container is normal destroyed.  Here is definitely not the case.